api_v1/app/app.py

```diff
@@ -0,0 +1,282 @@
+import sys
+sys.path.append('..')
+import re
+import json
+import datetime
+import jwt
+from functools import wraps
+from flask import Flask, request, session, jsonify, make_response, render_template
+from classes.user import User
+
+app = Flask(__name__)
+
+app.secret_key = 'younoscretet'
+
+''' creating an instance of the user '''
+user = User()
+
+''' dictionary to store our users'''
+users = {}
+
+def token_required (f):
+	""" decoding received token """
+	@wraps(f)
+	def decorate(*args, **kwargs):
+		token = None
+
+		if 'Authorization' in request.headers:
+			token = request.headers['Authorization']
+
+		if not token:
+			return jsonify({'message': 'Token is missing'}), 403
+		try:
+			payload = jwt.decode(token, app.secret_key)
+			current_user = payload['sub']
+
+		except jwt.ExpiredSignatureError:
+			return jsonify ({'message': 'Expired token!'}), 403
+
+		except jwt.InvalidTokenError:
+			return jsonify({'message': 'Invalid token!'}), 403
+
+		return f(current_user, *args, **kwargs)
+
+	return decorate
+
+
+@app.route('/')
+def display_documentation():
+	""" route displays api documentation """
+
+	return render_template('index.html')
+
+
+@app.route('/api/v1/auth/register', methods = ['POST'])
+def register():
+	""" route enables a user to register """
+
+	data = request.get_json()
+	if data:
+
+		username = data['username']
+		name_match = re.match('^[A-Za-z0-9]+$', username)
```

Comment:
Would you mind creating a decorator that does the `regex` matching for the given fields? Because it looks like we will have to repeat this code multiple times. This could look like
```python
@validate(username='[^[A-Za-z0-9]+$', password='^[A-Za-z0-9]{4,}$')
```
We can have a chat on slack if you need help on how to do this.